### PR TITLE
Add test for Temporal.Duration.from

### PR DESCRIPTION
**Pull request by FrankYFTang**
#3002  @jugglinmike @ryzokuken @Ms2ger

### CODE REVIEW

File: test/built-ins/Temporal/Duration/from/duration.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// Copyright (C) 2021 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+esid: sec-temporal.duration.from
+description: Temporal.Duration.from will crate a Duration from a given Duration.
```
Comment (rwaldron) [inline test/built-ins/Temporal/Duration/from/duration.js:6]: s/crate/create

File: test/built-ins/Temporal/Duration/from/duration.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  assert.sameValue(days, duration.days, "days");
+  assert.sameValue(hours, duration.hours, "hours");
+  assert.sameValue(minutes, duration.minutes, "minutes");
+  assert.sameValue(seconds, duration.seconds, "seconds");
+  assert.sameValue(milliseconds, duration.milliseconds, "milliseconds");
+  assert.sameValue(microseconds, duration.microseconds, "microseconds");
+  assert.sameValue(nanoseconds, duration.nanoseconds, "nanoseconds");
+  assert.sameValue(sign, duration.sign, "sign");
+  assert.sameValue(blank, duration.blank, "blank");
+}
```
Comment (rwaldron) [inline test/built-ins/Temporal/Duration/from/duration.js:27]: If this is going to be used many times, please add it to https://github.com/tc39/test262/blob/main/harness/temporalHelpers.js

File: test/built-ins/Temporal/Duration/from/string-throw-range-error.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+assert.throws(RangeError, () => Temporal.Duration.from("PT,1H"),
+    "Invalid time value");
+assert.throws(RangeError, () => Temporal.Duration.from("PT.1M"),
+    "Invalid time value");
+assert.throws(RangeError, () => Temporal.Duration.from("PT,1M"),
+    "Invalid time value");
+assert.throws(RangeError, () => Temporal.Duration.from("PT.1S"),
+    "Invalid time value");
+assert.throws(RangeError, () => Temporal.Duration.from("PT,1S"),
+    "Invalid time value");
```
Comment (rwaldron) [inline test/built-ins/Temporal/Duration/from/string-throw-range-error.js:56]: There are two options here: 

1. You can update these assert messages to be meaningful
2. I can run a script on them later and generate something more meaningful, but you'll have no say in them

File: test/built-ins/Temporal/Duration/from/string.js [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+assertDuration(Temporal.Duration.from("PT3,025M"),
+    0, 0, 0, 0, 0, 3, 1, 500, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,01M"),
+    0, 0, 0, 0, 0, 3, 0, 600, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,005M"),
+    0, 0, 0, 0, 0, 3, 0, 300, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,001M"),
+    0, 0, 0, 0, 0, 3, 0, 60, 0, 0, 1, false);
+assertDuration(Temporal.Duration.from("PT3,006M"),
+    0, 0, 0, 0, 0, 3, 0, 360, 0, 0, 1, false);
```
Comment (rwaldron) [inline test/built-ins/Temporal/Duration/from/string.js:179]: What happens when the wrong number of arguments are passed?

File: test/built-ins/Temporal/Duration/from/string-throw-range-error.js (hunk 3 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Duration/from/string-throw-range-error.js:56]: 2. is ok for me.

File: test/built-ins/Temporal/Duration/from/string.js (hunk 4 above)
Comment (FrankYFTang) [reply test/built-ins/Temporal/Duration/from/string.js:179]: "PT3,001M" in ISO8601 is a different way to represent "PT3.001M" , whic means 3.001 minutes.
which is 3 minutes and 60 milliseconds

"PT3,006M" in ISO8601 is a different way to represent "PT3.006M" , whic means 3.006 minutes.
which is 3 minutes and 360 milliseconds

See https://tc39.es/proposal-temporal/#prod-DecimalSeparator both "." and "," are DecimalSeparator .

If your question is about the number of argument of the from method- there is only one 
https://tc39.es/proposal-temporal/#sec-temporal.duration.from

**Comment by rwaldron** [comment]
@FrankYFTang the additions to `harness/temporalHelpers.js` are in `main` now, so you can rebase this and update it.

**Comment by ptomato** [comment]
I'd recommend not merging this as it's obviously testing a mistake in the spec text, and we intend to ask for consensus at the next TC39 meeting to change it.

**Comment by FrankYFTang** [comment]
ok, let me just abandon this PR and wait for @ptomato  and @Ms2ger  to move their test from Temporal repo instead.

**Pull request closed without merging**